File: lib/layer2_core/light/e_on_primary_container_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EOnPrimaryContainerLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl19;
      case 'standard': return cstl19;
      case 'blue': return cbl19;
      case 'green': return cgl19;
      case 'red': return crl19;
      case 'nothing': return cnol19;
      default: return cstl19;
    }
  }
}

File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EOnPrimaryContainerLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnPrimaryContainerLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl19
            AppTheme.STANDARD -> R.color.cstl19
            AppTheme.BLUE -> R.color.cbl19
            AppTheme.GREEN -> R.color.cgl19
            AppTheme.RED -> R.color.crl19
            AppTheme.NOTHING -> R.color.cnol19
        }
    }
}
